src/common/Metric/Metric.h

```diff
@@ -206,7 +209,9 @@ MetricStopWatch<LoggerType> MakeMetricStopWatch(LoggerType&& loggerFunc)
 #define TC_METRIC_DETAILED_TIMER(category, ...)                                                                  \
         MetricStopWatch TC_METRIC_UNIQUE_NAME(__tc_metric_stop_watch) = MakeMetricStopWatch([&](TimePoint start) \
         {                                                                                                        \
-            sMetric->LogValue(category, std::chrono::steady_clock::now() - start, { __VA_ARGS__ });              \
+            int64 duration = int64(std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::steady_clock::now() - start).count()); \
```

Comment:
You could use the `Milliseconds` typedef, optionally even the `TimePoint` typedef for `std::chrono::steady_clock`.

Reply:
is TimePoint::now() a thing ?